app/lib/features/space/sheets/link_room_sheet.dart

```diff
@@ -349,6 +357,62 @@ class _LinkRoomPageConsumerState extends ConsumerState<LinkRoomPage> {
       final room = await ref.watch(maybeRoomProvider(roomId).future);
       if (room != null) {
         room.addParentRoom(selectedParentSpaceId, true);
+        final joinRule = room.joinRuleStr();
```

Comment:
Better to have:
- Can we make separate function for `join logic` for code separation?

Reply:
yeah, I can do that.